pkg/genericapiserver/authorizer/authz.go

```diff
@@ -72,6 +72,29 @@ func NewAlwaysFailAuthorizer() authorizer.Authorizer {
 	return new(alwaysFailAuthorizer)
 }
 
+type adminAuthorizer struct {
+	user string
+	uid  string
+}
+
+func (r *adminAuthorizer) Authorize(attr authorizer.Attributes) (bool, string, error) {
+	if r.user != "" && attr.GetUser() != nil &&
+		attr.GetUser().GetName() == r.user &&
+		attr.GetUser().GetUID() == r.uid {
+		return true, "", nil
+	}
+	return false, "", errors.New("Authorization failure.")
+}
+
+// NewTokenAuthorizer is for use in loopback scenarios
+func NewTokenAuthorizer(user string, uid string) *adminAuthorizer {
```

Comment:
> In lines 81 through 85, i check fir the user name, uid etc. Do we not want those checks? Do we just check if the group is present in this list of groups that you are proposing? (NewPrivilegedGroups(groups ...string))

Yes.  The group is reserved for internal use (`system:` prefix) and keying off of that group will signed certificates to make use of it if they wish and will allow the RBACSuperUser to collapse into the same code path you're providing here.